app/actions/remote/thread.ts

```diff
@@ -1,13 +1,196 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {switchToThread} from '@actions/local/thread';
+import {markTeamThreadsAsRead, processReceivedThreads, switchToThread, updateThread} from '@actions/local/thread';
 import {fetchPostThread} from '@actions/remote/post';
+import {General} from '@constants';
+import DatabaseManager from '@database/manager';
+import NetworkManager from '@init/network_manager';
+import {getChannelById} from '@queries/servers/channel';
+import {getPostById} from '@queries/servers/post';
+import {getCommonSystemValues} from '@queries/servers/system';
+import {getIsCRTEnabled, getThreadById} from '@queries/servers/thread';
+
+import {forceLogoutIfNecessary} from './session';
+
+type FetchThreadsRequest = {
+    error?: unknown;
+} | {
+    data: GetUserThreadsResponse;
+};
+
+type FetchThreadsOptions = {
+    before?: string;
+    after?: string;
+    perPage?: number;
+    deleted?: boolean;
+    unread?: boolean;
+    since?: number;
+}
 
 export const fetchAndSwitchToThread = async (serverUrl: string, rootId: string) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+
     // Load thread before we open to the thread modal
-    // https://mattermost.atlassian.net/browse/MM-42232
+    // @Todo: https://mattermost.atlassian.net/browse/MM-42232
     fetchPostThread(serverUrl, rootId);
 
+    // Mark thread as read
+    const isCRTEnabled = await getIsCRTEnabled(database);
+    if (isCRTEnabled) {
+        const post = await getPostById(database, rootId);
```

Comment:
I wonder if this chain somehow gets easier if we use any of the relationships between models (if they are properly configured). i.e. `thread = await post.thread.fetch(); channel = await post.channel.fetch();`